lib/config/typescriptReact.js

```diff
@@ -0,0 +1,23 @@
+var merge = require('merge');
+
+module.exports = {
+  extends: [
+    // react should come before typescript
```

Comment:
`plugin:shopify/typescript`'s config has the disabled rules that override the standard `esnext` rules that conflict with the parser. 
Listing `plugin:shopify/typescript` last ensures that the disabled rules stay disabled, instead of being re-enabled by the react config.